common/lib/platform.sh

```diff
@@ -49,7 +51,45 @@ passthrough_envars() {
     done
 
     echo "Warning: Will pass env. vars. matching the following regex:
-$passthrough_env_re" > /dev/stderr
+$passthrough_env_re" >> /dev/stderr
 
     compgen -A variable | grep -Ev "$SECRET_ENV_RE" | grep -E  "$passthrough_env_re"
 }
+
+# On more occasions than we'd like, it's necessary to put temporary
+# platform-specific workarounds in place.  To help ensure they're
+# actually be temporary, it's useful to place a time-limit on them.
+# This function accepts two arguments:
+# - A (required) future, date/time of the form YYYYMMDD (UTC based).
+# - An (optional) message string to present after the 1st argument date
+timebomb() {
+    local expire="$1"
+
+    if ! expr "$expire" : '[0-9]\{8\}$' >> /dev/null; then
+      echo "timebomb: '$expire' must be UTC-based and of the form YYYYMMDD"
+      exit 1
+    fi
+
+    if [[ $(date -u +%Y%m%d) -lt $(date -u -d "$expire" +%Y%m%d) ]]; then
+        return
+    fi
+
+    declare -a frame
+    read -a frame < <(caller)
+
+    cat << EOF >> /dev/stderr
+***********************************************************
+* TIME BOMB EXPIRED!
+*
+*   >> ${frame[1]}:${frame[0]}: ${2:-No reason given, tsk tsk}
+*
+* Temporary workaround expired on '$expire'.
```

Comment:
Why the quotes? That's weird.

>Temporary workaround expired on 20240102

>Temporary workaround expired on '20240102'

Reply:
I did that in case $expire` is an empty string...but there's a check for that at the start of the function...so I'll drop the quotes then.  Thanks for catching this.

Reply:
Hmmm...actually, would you be opposed to me doing:

`* Temporary workaround expired on ${expire:0:4}-${expire:4:2}-${expire:6:2}.`

?